tests/types.ts

```diff
@@ -1,62 +1,126 @@
-        public_asset: 'Asset',
-        commitment: '[u8; 32]',
-    },
-    Asset: {
-        id: '[u8; 32]',
-        value: '[u8; 16]',
+  Checkpoint: {
+    receiver_index: "[u64; 256]",
+    sender_index: "u64",
+  },
+  FullIncomingNote: {
+    address_partition: "u8",
+    incoming_note: "IncomingNote",
+    light_incoming_note: "LightIncomingNote",
+  },
+  IncomingNote: {
+    ephemeral_public_key: "[u8; 32]",
+    tag: "[u8; 32]",
+    ciphertext: "[[u8;32]; 3]",
+  },
+  LightIncomingNote: {
+    ephemeral_public_key: "[u8; 32]",
+    ciphertext: "[[u8;32]; 3]",
+  },
+  Utxo: {
+    is_transparent: "bool",
+    public_asset: "Asset",
+    commitment: "[u8; 32]",
+  },
+  Asset: {
+    id: "[u8; 32]",
+    value: "[u8; 16]",
+  },
+  OutgoingNote: {
+    ephemeral_public_key: "[u8; 32]",
+    ciphertext: "[[u8;32]; 2]",
+  },
+  PullResponse: {
+    should_continue: "bool",
+    receivers: "Vec<(Utxo, FullIncomingNote)>",
+    senders: "Vec<([u8; 32], OutgoingNote)>",
+    senders_receivers_total: "[u8; 16]",
+  },
+  DensePullResponse: {
+    should_continue: "bool",
+    receivers: "String",
+    senders: "String",
+    senders_receivers_total: "[u8; 16]",
+    next_checkpoint: "Option<Checkpoint>",
+  },
+};
+
+export const rpc_api = {
+  mantaPay: {
+    pull_ledger_diff: {
+      description: "pull from mantaPay",
+      params: [
+        {
+          name: "checkpoint",
+          type: "Checkpoint",
+        },
+        {
+          name: "max_receivers",
+          type: "u64",
+        },
+        {
+          name: "max_senders",
+          type: "u64",
+        },
+      ],
+      type: "PullResponse",
     },
-    OutgoingNote: {
-        ephemeral_public_key: '[u8; 32]',
-        ciphertext: '[[u8;32]; 2]',
+    dense_pull_ledger_diff: {
+      description: "pull from mantaPay",
+      params: [
+        {
+          name: "checkpoint",
+          type: "Checkpoint",
+        },
+        {
+          name: "max_receivers",
+          type: "u64",
+        },
+        {
+          name: "max_senders",
+          type: "u64",
+        },
+      ],
+      type: "DensePullResponse",
     },
-    PullResponse: {
-        should_continue: 'bool',
-        receivers: 'Vec<(Utxo, FullIncomingNote)>',
-        senders: 'Vec<([u8; 32], OutgoingNote)>',
-        senders_receivers_total: '[u8; 16]',
-    }
+  },
 };
 
-export const rpc_api = {
-    mantaPay: {
-        pull_ledger_diff: {
-            description: 'pull from mantaPay',
-            params: [
-                {
-                    name: 'checkpoint',
-                    type: 'Checkpoint'
-                },
-                {
-                    name: 'max_receivers',
-                    type: 'u64'
-                },
-                {
-                    name: 'max_senders',
-                    type: 'u64'
-                }
-            ],
-            type: 'PullResponse'
-        }
-    }
-}
\ No newline at end of file
+const $Asset = $.object(
```

Comment:
Hope adding some summary about following batch codes to make next guys understand like:  A predefined tools to help decode a `PullResponse` from a raw bytes array.

Reply:
I have added some comments above the type definition.